### PR TITLE
PMM-12573 - Upgrade Podman UI

### PR DESCRIPTION
**Pull request by catalinaadam**

### CODE REVIEW

File: docs/pmm-upgrade/upgrade_podman.md [hunk 1]
```diff
@@ -5,48 +5,137 @@
...
+            After=network-online.target
+            After=nss-user-lookup.target nss-lookup.target
+            After=time-sync.target
 
-    ```sh
-    source ~/.config/pmm-server/env
-    podman pull ${PMM_IMAGE}:${PMM_TAG}
-    ```
+            [Service]
+            EnvironmentFile=/home/admin/.config/systemd/user/pmm-server.env
```
Comment (BupycHuk) [inline docs/pmm-upgrade/upgrade_podman.md:35]: replace `/home/admin` by `~`

File: docs/pmm-upgrade/upgrade_podman.md [hunk 2]
```diff
@@ -5,48 +5,137 @@
...
+
+            ExecStart=/usr/bin/podman run --rm --replace=true --name %N \
+                -v ${XDG_RUNTIME_DIR}/podman/podman.sock:/var/run/docker.sock \
+                -e WATCHTOWER_HTTP_API_UPDATE=${WATCHTOWER_HTTP_API_UPDATE} \
+                -e WATCHTOWER_HTTP_API_TOKEN=${WATCHTOWER_HTTP_API_TOKEN} \
+                -e WATCHTOWER_NO_RESTART=${WATCHTOWER_NO_RESTART} \
+                -e WATCHTOWER_DEBUG=${WATCHTOWER_DEBUG} \
+                --net pmm_default \
+                --cap-add=net_admin,net_raw \
+                docker.io/perconalab/watchtower:PMM-12573-podman
```
Comment (BupycHuk) [inline docs/pmm-upgrade/upgrade_podman.md:89]:
```suggestion
                docker.io/perconalab/watchtower:latest
```

File: docs/pmm-upgrade/upgrade_podman.md (hunk 1 above)
Comment (BupycHuk) [reply docs/pmm-upgrade/upgrade_podman.md]: in other places too

File: docs/pmm-upgrade/upgrade_podman.md [hunk 3]
```diff
@@ -5,48 +5,138 @@
...
+
+    Follow these steps to upgrade your PMM Server while preserving your monitoring data and settings—you can restore from your backup if needed.
+    {.power-number}
+
+    1. [Back up your data](../install-pmm/install-pmm-server/baremetal/podman/backup_container_podman.md).
+
+    2. Update PMM tag by editing `~/.config/systemd/user/pmm-server.env` file and running the following command to set the latest release version:
+   
+        ```sh
+        sed -i "s/PMM_TAG=.*/PMM_TAG=3.0.0/g" ~/.config/systemd/user/pmm-server.env
```
Comment (BupycHuk) [inline docs/pmm-upgrade/upgrade_podman.md:116]:
```suggestion
        sed -i "s/PMM_IMAGE=.*/PMM_IMAGE=3.0.0/g" ~/.config/systemd/user/pmm-server.env
```

File: docs/pmm-upgrade/upgrade_podman.md [hunk 4]
```diff
@@ -5,48 +5,137 @@
...
-2. Verify your current PMM version: Check your current PMM version by navigating to **PMM Configuration > Updates** or by running the following command. If accessing remotely, replace `localhost` with the appropriate IP address or server name:
+2. Verify your current PMM version from  **PMM Configuration > Updates** or by running the following command. If accessing remotely, replace `localhost` with the appropriate IP address or server name:
 
     ```sh
     podman exec -it pmm-server \
     curl -ku admin:admin https://localhost/v1/version
     ```
 
-## Upgrade steps
+You can upgrade PMM Server using Podman either through the UI or manually using Podman commands:
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply docs/pmm-upgrade/upgrade_podman.md:17]: left a comment in Jira

File: docs/pmm-upgrade/upgrade_podman.md [hunk 5]
```diff
@@ -20,16 +20,16 @@ Follow these steps to upgrade your PMM Server while preserving your monitoring d
 {.power-number}
 
 1. [Back up your data](../install-pmm/install-pmm-server/baremetal/podman/backup_container_podman.md).
+2. Update PMM tag by editing `~/.config/systemd/user/pmm-server.env` file and running the following command to set the latest release version:
 
-2. Update PMM tag by editing `~/.config/pmm-server/env` file and running the following command to set the latest release version:
     ```sh
-    sed -i "s/PMM_TAG=.*/PMM_TAG=3.0.0/g" ~/.config/pmm-server/env
-    ```
+    sed -i "s/PMM_IMAGE=.*/PMM_IMAGE=docker.io/pmm-server:3.0.0/g" ~/.config/systemd/user/pmm-server.env
```
Comment (BupycHuk) [inline docs/pmm-upgrade/upgrade_podman.md:26]:
```suggestion
    sed -i "s/PMM_IMAGE=.*/PMM_IMAGE=docker.io/percona/pmm-server:3.0.0/g" ~/.config/systemd/user/pmm-server.env
```

File: docs/pmm-upgrade/upgrade_podman.md [hunk 6]
```diff
@@ -45,8 +45,8 @@ Follow these steps to upgrade your PMM Server while preserving your monitoring d
     podman ps | grep pmm-server
     ```
 
-3. Check the logs for any errors:
+6. Check the logs for any errors:
 
     ```sh
     podman logs pmm-server
-    ```
+    ```
```
Comment (BupycHuk) [inline docs/pmm-upgrade/upgrade_podman.md:52]: should we explain mechanism how we upgrade podman or it's not important?

File: docs/install-pmm/install-pmm-server/baremetal/podman/index.md [hunk 7]
```diff
@@ -26,125 +27,144 @@ Percona recommends running PMM as a non-privileged user and running it as part o
...
+        ```sh
+        systemctl --user enable --now pmm-server
+        systemctl --user enable --now watchtower
+        ```
+
+    5. Go to `https://localhost:8443` to access the PMM user interface in a web browser. If you are accessing the host remotely, replace `localhost` with the IP or server name of the host.
+
+=== "Installation with manual updates"
+
+    The installation with manual updates offers a straightforward setup with direct control over updates, without relying on additional services. In this approach, you manually update the `PMM_TAG` in the environment file and restart the PMM Server service. SystemD then automatically manages the container replacement.
```
Comment (BupycHuk) [inline docs/install-pmm/install-pmm-server/baremetal/podman/index.md:125]:
```suggestion
    The installation with manual updates offers a straightforward setup with direct control over updates, without relying on additional services. In this approach, you manually update the `PMM_IMAGE` in the environment file and restart the PMM Server service. SystemD then automatically manages the container replacement.
```

File: docs/install-pmm/install-pmm-server/baremetal/podman/index.md [hunk 8]
```diff
@@ -26,125 +27,144 @@ Percona recommends running PMM as a non-privileged user and running it as part o
...
+        Wants=network-online.target
+        After=network-online.target
+        After=nss-user-lookup.target nss-lookup.target
+        After=time-sync.target
+        [Service]
+        EnvironmentFile=~/.config/systemd/user/pmm-server.env
+        Restart=on-failure
+        RestartSec=20
+        ExecStart=/usr/bin/podman run \
+            --volume ~/.config/systemd/user/:/home/pmm/update/ \
```
Comment (BupycHuk) [inline docs/install-pmm/install-pmm-server/baremetal/podman/index.md:142]:
```suggestion
```

File: docs/install-pmm/install-pmm-server/baremetal/podman/index.md [hunk 9]
```diff
@@ -26,125 +27,144 @@ Percona recommends running PMM as a non-privileged user and running it as part o
...
+        ExecStop=/usr/bin/podman stop -t 10 %N
+        [Install]
+        WantedBy=default.target
+        ```
+
+    2. Create the environment file at `~/.config/systemd/user/pmm-server.env`:
+   
+        ```sh
+        PMM_WATCHTOWER_HOST=http://watchtower:8080
+        PMM_WATCHTOWER_TOKEN=123
```
Comment (BupycHuk) [inline docs/install-pmm/install-pmm-server/baremetal/podman/index.md:158]:
```suggestion
```
it's not required

**Review by BupycHuk — APPROVED** [review]

**Review by peterSirotnak — APPROVED** [review]

**Pull request merged**